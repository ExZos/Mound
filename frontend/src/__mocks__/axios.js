import { endpoints as api } from '../endpoints';

const axios = {
  create: () => axios,
  get: (url) => {
    switch(url) {
      case api.getSpaceByName + "Headon's Floor/":
        return Promise.resolve({
          data: {
            id: 1,
            name: "Headon's Floor",
            status: true
          }
        });
        break;

      default:
        return Promise.reject();
    }
  }
};

export default axios;
