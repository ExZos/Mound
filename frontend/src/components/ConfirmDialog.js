import React from "react";
import { Button } from "reactstrap";
import { Dialog, DialogTitle, DialogContent, DialogActions, IconButton } from '@material-ui/core';

import CloseIcon from '@material-ui/icons/Close';

import GeneralComponent from './GeneralComponent';
import '../styles/confirmDialog.css';

class ConfirmDialog extends GeneralComponent {
  confirm = () => {
    this.props.toggleModal();

    this.props.confirm();
  }

  render() {
    return (
      <Dialog id="confirmDialog" open={this.props.showModal}>
        <DialogTitle onClose={this.props.toggleModal}>
          {this.props.mHeader}

          <IconButton className="closeDialog" aria-label="close" onClick={this.props.toggleModal}>
            <CloseIcon />
          </IconButton>
        </DialogTitle>

        <DialogContent dividers>
          {this.props.mBody}
        </DialogContent>

        <DialogActions>
          <Button className="modalConfirm" color="primary" onClick={this.confirm}>Confirm</Button>
          <Button className="modalCancel" color="secondary" onClick={this.props.toggleModal}>Cancel</Button>
        </DialogActions>
      </Dialog>
    );
  }
}

export default ConfirmDialog;
