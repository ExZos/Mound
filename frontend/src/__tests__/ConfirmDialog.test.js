import React from 'react';
import { shallow, mount, render } from 'enzyme';
import axios from 'axios';
import { createBrowserHistory } from 'history';

import ConfirmDialog from '../components/ConfirmDialog';

const history = createBrowserHistory();

describe('ConfirmDialog', () => {
  it('should render correctly with props', () => {
    const component = shallow(<ConfirmDialog showModal={true}
      setShowDialog={jest.fn()} confirm={jest.fn()}
      mHeader='Some header' mBody='Some body'
    />);
    const dialogTitle = component.find('WithStyles(ForwardRef(DialogTitle))');
    const dialogContent = component.find('WithStyles(ForwardRef(DialogContent))');

    expect(component).toMatchSnapshot();
    expect(dialogTitle.childAt(0).text()).toEqual('Some header');
    expect(dialogContent.childAt(0).text()).toEqual('Some body');
  });

  it('should call cancel function prop on cancel click', () => {
    const mockCancel = jest.fn();
    const mockConfirm = jest.fn();

    const component = shallow(<ConfirmDialog showDialog={true}
      setShowDialog={mockCancel} confirm={mockConfirm}
      mHeader='Some header' mBody='Some body'
    />);
    const button = component.find('.modalCancel');

    button.props().onClick();

    expect(mockCancel).toHaveBeenCalled();
    expect(mockConfirm).not.toHaveBeenCalled();
  });

  it('should call confirm function prop on confirm click', () => {
    const mockCancel = jest.fn();
    const mockConfirm = jest.fn();

    const component = shallow(<ConfirmDialog showDialog={true}
      setShowDialog={mockCancel} confirm={mockConfirm}
      mHeader='Some header' mBody='Some body'
    />);
    const button = component.find('.modalConfirm');

    button.props().onClick();

    expect(mockConfirm).toHaveBeenCalled();
    expect(mockCancel).toHaveBeenCalled();
  });

  it('should show the modal', () => {
    const component = shallow(<ConfirmDialog showDialog={true}
      setShowDialog={jest.fn()} confirm={jest.fn()}
      mHeader='Some header' mBody='Some body'
    />);

    expect(component.props()).toHaveProperty('open', true);
  });

  it('should hide the modal', () => {
    const component = shallow(<ConfirmDialog showDialog={false}
      setShowDialog={jest.fn()} confirm={jest.fn()}
      mHeader='Some header' mBody='Some body'
    />);

    expect(component.props()).toHaveProperty('open', false);
  });
});
