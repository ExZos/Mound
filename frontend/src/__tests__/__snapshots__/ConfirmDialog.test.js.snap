// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog should render correctly with props 1`] = `
<WithStyles(ForwardRef(Dialog))
  id="confirmDialog"
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    onClose={[MockFunction]}
  >
    Some header
    <WithStyles(ForwardRef(IconButton))
      aria-label="close"
      className="closeDialog"
      onClick={[MockFunction]}
    >
      <CloseIcon />
    </WithStyles(ForwardRef(IconButton))>
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))
    dividers={true}
  >
    Some body
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <Button
      className="modalConfirm"
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      Confirm
    </Button>
    <Button
      className="modalCancel"
      color="secondary"
      onClick={[MockFunction]}
      tag="button"
    >
      Cancel
    </Button>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
